### DISCUSSION TITLE
Any way to use current timestamp operator properly?

### DISCUSSION THREAD

**Discussion by neos1803** [vitaly-t/pg-promise#925] [Q&A] [answered]
Hello. I came upon a bug and it's related to current_timestamp operator provided by postgresql. Sometimes it's working, but sometimes it's not. I decided to make a simple test and it seems like the query that uses current_timestamp doesn't return any result. Weird enough, because it returns a result when it's run on database admin tools like dbeaver.

Here's the snippet of the test script

![image](https://github.com/vitaly-t/pg-promise/assets/38176579/5d5b230f-f0af-4406-98a4-01ef4ea3aa48)

This is the result when i run it

![image](https://github.com/vitaly-t/pg-promise/assets/38176579/2e20a853-823f-4433-af31-15da0f96f322)

This is the result when i run the query which use current timestamp in database tool
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/84752bce-1fdb-49e0-9e52-253885b62f00)
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/cd4986f4-9f6b-456e-95be-361b553d1c4a)

Is there anything i miss?

By the way i am using pg-promise 11.5.4 and my pg version is 15.6.

**Comment by neos1803** [comment]
After digging a bit, it seems like there's a different timezone between my script and my database.

Script
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/18074a15-0877-40fc-a38f-46d5cc7264c6)

Database
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/3ac6a9f5-3164-40a0-95eb-76dc2b444a92)
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/7fdd1cf0-3683-4916-8f01-323b33de163b)

However, i have try forcing it in the query to convert it to use Asia/Jakarta as timezone, but out of 3 queries, only one returns the same value as the database version (the one that converts it to string, instead of date in javascript)

Script
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/85ab656d-e482-43f5-8e3f-3e8e63e0540c)
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/ee2bfcb8-9388-4096-b323-e16de69da4d4)

Database
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/ed9c3268-3a16-4f26-9a34-667b6469ab54)
![image](https://github.com/vitaly-t/pg-promise/assets/38176579/92a60bd8-6d58-4a85-9a7c-e929283afbd8)

**Comment by vitaly-t** [comment]
Timezones can be tricky sometimes...

* There are time parsers that you can set via `pgp.pg.types.setTypeParser(1114, ...`
* You can also set timezone dynamically, via `SET TIME ZONE`

You should check for those things, based on your case. There are plenty of examples on the internet to manage those, based on case scenario.

**Reply by neos1803** [reply, marked as answer]
It seems like i need to use type assertion on  duedate field so it can match current_timestamp.  Duedate is timestampz while current_timestamp is timestamp without timezone. As you say, timezones is indeed tricky. Thanks for replying.
